ceasiompy/ActuatorDisk/func/optimalprop.py

```diff
@@ -0,0 +1,416 @@
+            advanced_ratio (float): Free_stream_velocity/(rotational_velocity*diameter)[-]
+            free_stream_velocity (float): Cruise velocity [m/s]
+            prandtl (bool): Correction for tip losses
+            blades_number (int): Blades propeller number[-]
+
+    Returns:
+        dCt_optimal (float): thrust coefficient at every radius [-]
+        dCp (float): power coefficient at every radius [-]
+        r (float): adimensional radius [-]
+    """
+
+    # Resize the vectors using the number of radial stations.
+    r = np.empty(stations)
+    non_dimensional_radius = np.empty(stations)
+    dCp = np.empty(stations)
+    induced_velocity_distribution = np.empty(stations)
+    new_axial_interference_factor = np.empty(stations)
+    old_axial_interference_factor = np.empty(stations)
+    initial_axial_interference_factor = np.empty(stations)
+    dCt_new = np.empty(stations)
+    dCt_old = np.empty(stations)
+    dCt_0 = np.empty(stations)
+    delta_pressure = np.empty(stations)
+    correction_function = np.empty(stations)
+    optimal_axial_interference_factor = np.empty(stations)
+    optimal_rotational_interference_factor = np.empty(stations)
+    dCt_optimal = np.empty(stations)
+
+    non_dimensional_hub_radius = hub_radius / radius
+
+    # Computation of the non-dimensional radial stations.
+    for i in range(1, stations + 1):
+        r[i - 1] = i / float(stations)
+        if r[i - 1] <= non_dimensional_hub_radius:
+            i_hub = i - 1
+
+    diameter = 2 * radius
+    n = free_stream_velocity / (diameter * advanced_ratio)
+    omega = n * 2 * pi
+
+    # Computation of the tip loss Prandtl correction function
+    if prandtl:
+        for i in range(stations):
+            correction_function[i] = (2 / pi) * acos(
+                exp(
+                    -0.5
+                    * blades_number
+                    * (1 - r[i])
+                    * sqrt(1 + (omega * radius / free_stream_velocity) ** 2)
+                )
+            )
+
+    else:
+        correction_function[stations] = 1.0
+
+    # Computation of the non-dimensional radius
+    for i in range(stations):
+        non_dimensional_radius[i] = omega * r[i] * radius / free_stream_velocity
+
+    EPSILON = 5e-20
+
+    radial_stations_spacing = 1.0 / stations
+
+    # Computation of the first try induced velocity distribution.
+    for i in range(stations):
+        induced_velocity_distribution[i] = (2 / free_stream_velocity**2) * (
+            (-1 / free_stream_velocity)
+            + sqrt(
+                1
+                + (
+                    (diameter**4 * (total_thrust_coefficient) * n**2)
+                    / (free_stream_velocity**2 * pi * r[i])
+                )
+            )
+        )
+
+    # Computation of the first try Lagrange moltiplicator.
+    first_lagrange_moltiplicator = 0.0
+    for i in range(stations):
+        first_lagrange_moltiplicator += induced_velocity_distribution[i]
```

Comment:
Is it not the equivalent of doing:
```suggestion
    first_lagrange_moltiplicator = np.sum(induced_velocity_distribution)
```
??

Reply:
https://www.pythonlikeyoumeanit.com/Module3_IntroducingNumpy/VectorizedOperations.html